c2corg_ui/static/js/yeti/yeti.js

```diff
@@ -0,0 +1,447 @@
+   * @private
+   */
+  this.timeout_ = $timeout;
+
+  /**
+   * @type {app.Alerts}
+   * @private
+   */
+  this.alerts_ = appAlerts;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.yetiUrlBase_ = 'http://yeti.ensg.eu/wps?request=Execute&service=WPS&version=1.0.0&identifier=Yeti&datainputs=';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.yetiUrl_ = '';
+
+  /**
+   * @type {?ol.layer.Image}
+   * @private
+   */
+  this.yetiLayer_ = null;
+
+  /**
+   * @type {?ol.Map}
+   * @private
+   */
+  this.map_ = null;
+  this.scope_.$root.$on('responseMap', this.setMap_.bind(this));
+  this.timeout_(() => {
+    this.scope_.$root.$emit('requestMap');
+  });
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.formOK = false;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.formOKError_ = 'bra';
+
+  /**
+   * @type {Object}
+   * @private
+   */
+  this.errors_ = {
+    'methode': {
+      'simple': 'Méthode manquante',
+      'full': 'Veuillez sélectionner une méthode pour le calcul'
+    },
+    'methode_bra': {
+      'simple': 'Méthode et BRA incompatible',
+      'full': 'La méthode MRD (débutant) est autorisée avec un BRA de 3 maximum'
+    },
+    'bra': {
+      'simple': 'BRA manquant',
+      'full': 'La valeur de BRA est manquante'
+    },
+    'altitude': {
+      'simple': 'Altitude manquante',
+      'full': 'L\'altitude est requise quand le BRA haut et bas sont différents'
+    },
+    'zoom': {
+      'simple': 'Zoom carte trop important',
+      'full': 'Veuillez zoomer au niveau ' + this.MIN_ZOOM + ' minimum'
+    },
+    'ok': 'Tout semble correct :)',
+    'yeti': 'Le service ne fonctionne pas actuellement',
+    'yeti_prefix': 'YETI Service: ',
+    'yeti_unauthorized': 'Vous devez être autorisé pour effectuer cette requête. Contactez jbl77 si vous êtes intéressé.'
+  };
+
+  /**
+   * @type {string}
+   * @export
+   */
+  this.actualError = '';
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.isProcessing = false;
+
+  /**
+   * @type {app.Authentication}
+   * @private
+   */
+  this.auth_ = appAuthentication;
+
+};
+
+/**
+ * Set initial scope data
+ * @private
+ */
+app.YetiController.prototype.setInitialData_ = function() {
+  // MRD
+  this.scope_['bra'] = {};
+  // MRE
+  this.scope_['rdv'] = {'none': true};
+  // MRP
+  this.setDanger_();
+  this.scope_['neigeMouillee'] = false;
+  this.scope_['tailleGroupe'] = 1;
+
+  // map
+  this.scope_['mapZoomOK'] = true;
+
+  // watch every input
+  this.scope_.$watchGroup([
+    'method',
+    'bra.haut',
+    'bra.bas',
+    'bra.altiseuil',
+    'bra.isDifferent',
+    'mapZoomOK'
+  ], (newValues, oldValues, scope) => {
+    // verif form
+    if (!this.scope_['bra']['haut']) {
+      this.formOK = false;
+      this.formOKError_ = 'bra';
+    } else if (!this.scope_['method']) {
+      this.formOK = false;
+      this.formOKError_ = 'methode';
+    } else if (
+      this.scope_['bra']['haut'] === 4 &&
```

Comment:
```suggestion
      this.scope_['bra']['haut'] > 3 &&
```

4 (or 3) could be a constant.